File: oop/src/Human.java
public class Human {
    private String fio;
    private int age;
    private double speed;
    private double x;
    private double y;

    // Конструктор
    public Human(String fio, int age, double speed, double x, double y) {
        this.fio = fio;
        this.age = age;
        this.speed = speed;
        this.x = x;
        this.y = y;
    }

    // Метод случайного шага (Random Walk)
    public void move() {
        double angle = Math.random() * 2 * Math.PI;
        setX(getX() + getSpeed() * Math.cos(angle));
        setY(getY() + getSpeed() * Math.sin(angle));
    }

    // Геттеры
    public String getFio() { return fio; }
    public int getAge() { return age; }
    public double getSpeed() { return speed; }
    public double getX() { return x; }
    public double getY() { return y; }

    // Сеттеры
    public void setX(double x) { this.x = x; }
    public void setY(double y) { this.y = y; }
    public void setSpeed(double speed) { this.speed = speed; }
}

File: visualka/src/main/kotlin/Human.kt
package org.example

import kotlin.random.Random

open class Human(
    protected val fullName: String,
    private val age: Int,
    protected var currentSpeed: Double,
    protected var x: Double = 0.0,
    protected var y: Double = 0.0
) {

    fun getFullName(): String = fullName

    fun getAge(): Int = age

    fun getCurrentSpeed(): Double = currentSpeed

    fun getX(): Double = x

    fun getY(): Double = y

    fun setCurrentSpeed(newSpeed: Double) {
        currentSpeed = newSpeed
    }

    open fun move(timeStep: Double) {
        val randomAngle = Random.nextDouble(0.0, 360.0)

        val distance = currentSpeed * timeStep

        val radians = Math.toRadians(randomAngle)

        val dx = distance * Math.cos(radians)
        val dy = distance * Math.sin(radians)

        x += dx
        y += dy

        println("${fullName} made a random step.")
        println(" -> New coordinates: ($x, $y)")
    }
}